crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -167,17 +167,21 @@ static int s2n_composite_cipher_aes_sha_encrypt(struct s2n_session_key *key, str
     POSIX_ENSURE_EQ(out->size, in->size);
 
     POSIX_GUARD_OSSL(EVP_EncryptInit_ex(key->evp_cipher_ctx, NULL, NULL, NULL, iv->data), S2N_ERR_KEY_INIT);
-    POSIX_GUARD_OSSL(EVP_Cipher(key->evp_cipher_ctx, out->data, in->data, in->size), S2N_ERR_ENCRYPT);
+
+    int len = out->size;
+    POSIX_GUARD_OSSL(EVP_EncryptUpdate(key->evp_cipher_ctx, out->data, &len, in->data, in->size), S2N_ERR_ENCRYPT);
+    S2N_ERROR_IF(len != in->size, S2N_ERR_ENCRYPT);
```

Comment:
Why are we adding this error check?

Reply:
Other cipher impls also have this check since EVP_EncryptUpdate returns the `len` encrypted https://github.com/aws/s2n-tls/blob/main/crypto/s2n_stream_cipher_rc4.c#L46

Reply:
`S2N_ERR_ENCRYPT` is more specific than `S2N_ERR_SAFETY`, which is kind of the [default error code if one is not provided](https://github.com/aws/s2n-tls/blob/fb1e57c104eea441f0a928bfa15eca6979810b00/utils/s2n_safety_macros.h#L86).

All error codes: https://github.com/aws/s2n-tls/blob/fb1e57c104eea441f0a928bfa15eca6979810b00/error/s2n_errno.h#L69

Reply:
I guess there's small inconsistency in the error reporting? Probably a fix for another pr

Reply:
Important to note: they also have different types https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L159-L162